src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -299,6 +299,15 @@ private void RunAssembly(string assemblyPath, TestFilter filter)
                 // we skip the native c++ binaries that we don't support.
                 TestLog.Warning("   Assembly not supported: " + assemblyPath);
             }
+            catch( NUnitEngineException e )
+            {
+                if( e.InnerException is BadImageFormatException )
+                {
+                    // we skip the native c++ binaries that we don't support.
+                    TestLog.Warning( "   Assembly not supported: " + assemblyPath );
+                }
+                throw;
```

Comment:
This `throw;` means that NUnitEngineExceptions are no longer handled at all. They used to be handled in the `catch (Exception)` section below.

Reply:
Fixed by #804